decidim-core/app/models/decidim/participatory_process_user_role.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+module Decidim
+  # Defines a relation between a user and a participatory process, and what
+  # kind of relation does the user has.
+  class ParticipatoryProcessUserRole < ApplicationRecord
+    belongs_to :user, foreign_key: "decidim_user_id", class_name: Decidim::User
+    belongs_to :participatory_process, foreign_key: "decidim_participatory_process_id", class_name: Decidim::ParticipatoryProcess
+
+    ROLES = %w(admin).freeze
+    validates :role, inclusion: { in: ROLES }
```

Comment:
Nope! `ParticipatoryProcessUserRole` should actually live *outside* core (didn't see this, but I think it should definitely live in `admin`). No one else needs that outside the admin panel.